build/keycloak/Dockerfile

```diff
@@ -1,41 +1,56 @@
-FROM jboss/keycloak:15.1.1
+##################################################
+# Step 1 - Build an optimized image
+##################################################
+FROM quay.io/keycloak/keycloak:20.0.3 as builder
+
+# These options can be modified to produce a different
+# optimized build.
+#
+# See  https://www.keycloak.org/server/containers
+# for more details.
+ENV KC_METRICS_ENABLED=true
+ENV KC_HEALTH_ENABLED=true
+ENV KC_FEATURES=preview
+ENV KC_DB=postgres
+ENV KC_HTTP_RELATIVE_PATH=/auth
```

Comment:
I kept the `/auth` relative path prefix so that folks can upgrade without breaking existing integrations/links.